hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/impl/ReactiveCollectionRecreateAction.java

```diff
@@ -0,0 +1,79 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.engine.impl;
+
+import java.io.Serializable;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.HibernateException;
+import org.hibernate.action.internal.CollectionAction;
+import org.hibernate.collection.spi.PersistentCollection;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.event.service.spi.EventListenerGroup;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.event.spi.PostCollectionRecreateEvent;
+import org.hibernate.event.spi.PostCollectionRecreateEventListener;
+import org.hibernate.event.spi.PreCollectionRecreateEvent;
+import org.hibernate.event.spi.PreCollectionRecreateEventListener;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.reactive.engine.ReactiveExecutable;
+import org.hibernate.reactive.persister.collection.impl.ReactiveCollectionPersister;
+
+public class ReactiveCollectionRecreateAction extends CollectionAction implements ReactiveExecutable {
```

Comment:
It seems to me that this should extend `CollectionRecreateAction`, to be consistent with what we do for entity-related actions, for example:

    public class ReactiveEntityDeleteAction extends EntityDeleteAction implements ReactiveExecutable

And that way we wouldn't need the copy/pasted `preRecreate()`/`postRecreate()` methods here.

Reply:
This will require a change to the ORM actions which are currently marked as ```final```.  I'll test locally and create a PR that'll have to be back-ported to 5.4 at some point.